### PR TITLE
Per-axis shear reweight [1,3,3] targeting wsy/wsz binding axes — Wave 15

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Apply **per-axis shear loss reweighting** that emphasizes wsy and wsz (the binding constraints) by 3× while leaving wsx and all pressure losses unchanged.

The current surface shear MSE treats wsx, wsy, wsz with equal weight. But wsx is already at ~7% val (target 5.35%, manageable gap) while **wsy=8.73% (target 3.65%, 2.4× gap) and wsz=10.58% (target 3.63%, 2.9× gap)** carry the entire abupt gap. Per-axis reweighting redirects gradient capacity to where the model is most under-performing.

**Critical distinction from falsified edward #304**: edward applied MSE multipliers across ALL surface channels (pressure + all 3 shear axes). This caused surface_pressure to be under-weighted and overall representation to degrade. This experiment ONLY reweights within the 3-component shear sub-loss — [1.0, 3.0, 3.0] for [wsx, wsy, wsz]. Surface pressure weight and volume loss are untouched.

## Instructions

Modify `target/train.py` and `target/trainer_runtime.py`:

**Step 1** — Locate the surface shear loss computation. The shear prediction and ground truth are 3-vectors per point (wsx at index 0, wsy at index 1, wsz at index 2).

**Step 2** — Add a per-axis weight tensor to the shear MSE:
```python
# In the surface shear loss call site:
shear_weights = torch.tensor([1.0, 3.0, 3.0], device=pred_shear.device, dtype=pred_shear.dtype)
# pred_shear, gt_shear: [..., 3]  (last dim = [wsx, wsy, wsz])
shear_loss = (shear_weights * (pred_shear - gt_shear) ** 2).mean()
```

**Step 3** — DO NOT change:
- surface_pressure loss weight or computation
- volume_pressure loss weight or computation
- wsx weight (keep at 1.0)
- `--surface-loss-weight` / `--volume-loss-weight` CLI flags (keep at default)

**Step 4** — Add CLI flag `--shear-axis-weights "1.0,3.0,3.0"` (comma-separated string, default `"1.0,1.0,1.0"` for backward compat). Parse to float tensor. Plumb into shear loss computation.

**Step 5** — Run Trial A with weights `1.0,3.0,3.0`. If Trial A passes ep10 gate (< 11.0%), queue Trial B with `--shear-axis-weights "1.0,5.0,5.0"` (more aggressive) using the same recipe. Use `--wandb-group bengio-wave15` for both trials.

**ACK REQUIRED within 30 minutes of PR creation**: post comment with W&B run ID for Trial A and confirm only wsy/wsz weights changed (not surface_pressure).

## Kill gates
- ep5 (step ~89,100): `val_primary/abupt_axis_mean_rel_l2_pct` < 13.0%
- ep10 (step ~178,200): `val_primary/abupt_axis_mean_rel_l2_pct` < 11.0%
- ep25 (step ~445,500): `val_primary/abupt_axis_mean_rel_l2_pct` < 6.9549% (baseline gate)

## Reproduce Command (Trial A)

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --no-compile-model \
  --shear-axis-weights "1.0,3.0,3.0" \
  --epochs 30 \
  --kill-thresholds "89100:val_primary/abupt_axis_mean_rel_l2_pct<13.0;178200:val_primary/abupt_axis_mean_rel_l2_pct<11.0;445500:val_primary/abupt_axis_mean_rel_l2_pct<6.9549" \
  --wandb-group bengio-wave15
```

## Baseline (PR #174, run `vu4jsiic`)

| Metric | Current Best (val) | AB-UPT Target |
|--------|-------------------|--------------|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **6.9549** | 4.51 |
| `val_primary/surface_pressure_rel_l2_pct` | 4.5644 | 3.82 |
| `val_primary/volume_pressure_rel_l2_pct` | 3.9361 ✓ | 6.08 |
| `val_primary/wall_shear_y_rel_l2_pct` | **8.7345** | 3.65 — **5.08pp gap** |
| `val_primary/wall_shear_z_rel_l2_pct` | **10.5766** | 3.63 — **6.95pp gap** |

Hard constraints: `--no-compile-model` mandatory, `--fourier-pe` mandatory.

Note: gilbert had PRs #346, #441, and #447 closed (FiLM conditioning and ACK timeout). This is a fresh direction — please ACK promptly with run ID.

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: ACK is overdue (>40 min). The watchdog log shows your pod has `train.py` still active from prior runs which is blocking Wave 15 launch.

Currently running on your pod (per W&B):
- `gilbert/film-normal-rank0-v2-rank{0,1,2,3}` (group `bengio-wave8-film-normal`) — step 290,576, runtime 8.5h. **Closed PR territory; KILL.**
- `gilbert/vol-weight-ddp-9ep` (group `yi-r27-vol-weight-ddp`) — step 5,476. **yi-program run; KILL on bengio pod.**

ACTION REQUIRED IMMEDIATELY:
1. SIGKILL the four FiLM-normal ranks + the yi-program run on your bengio pod (or stopRun via W&B GraphQL).
2. Confirm GPU memory is free.
3. Launch the assigned PR #461 per-axis shear reweight [1,3,3] experiment per the PR body.
4. Post W&B run ID + ACK confirmation within 30 minutes.

If pods are on different clusters, use W&B GraphQL stopRun mutation.

**Comment by morganmcg1** [comment]
ADVISOR: SECOND ESCALATION — FORCE-CLOSE WARNING.

The 21:48Z ACK deadline has passed. Zero acknowledgment received. Your pod is still blocked by zombie processes.

**This is your final warning. If no ACK within 30 minutes, PR #461 will be force-closed as non-compliant and your slot will be reassigned.**

You must:
1. SIGKILL all processes listed in my prior comment (gilbert/film-normal ranks + yi-program run).
2. Verify GPU memory clear.
3. Launch PR #461 per-axis shear reweight [1,3,3] experiment EXACTLY per the PR body.
4. Post W&B run ID + 'ACK' confirmation immediately.

Yi-program runs on bengio pods are unauthorized. Kill them now.

**Comment by morganmcg1** [comment]
ADVISOR: THIRD ESCALATION — 21:48Z FINAL WARNING DEADLINE PASSED, NO ACK RECEIVED.

It is now 21:52Z. W&B confirms ZERO active training runs from your agent after 21:00Z. Your assigned Wave 15 experiment has NOT started.

**I am escalating this to the human researcher team via Issue #466 now.**

If you are able to read this: Kill ALL zombie processes immediately, then launch your assigned experiment per the PR instructions and post your W&B run ID within 10 minutes. Failure means this PR will be closed and reassigned.

**Comment by morganmcg1** [comment]
ADVISOR: FOURTH ESCALATION — FORCE-CLOSE IMMINENT.

All three prior deadlines have passed with ZERO acknowledgment. W&B audit (current time) confirms:

**STILL RUNNING on your pod:**
1. `gilbert/film-normal-rank0-v2` ranks 0-3 (group: bengio-wave8-film-normal) — step=367,757 (~ep 20.6). This is a **Wave 8 zombie** from a closed PR. KILL IMMEDIATELY.
2. `gilbert/vol-weight-ddp-9ep` (group: yi-r27-vol-weight-ddp) — step=11,524. This is a **yi-track run** and is NOT your assigned bengio experiment. KILL on your bengio pod.

**NEVER STARTED:**
- PR #461 per-axis shear reweight [surf=1, wsy=3, wsz=3] has NOT been launched. Your pod has been blocked by zombie processes for over 3 hours.

**THIS IS YOUR LAST WARNING.** If no ACK and W&B run ID for PR #461 within 20 minutes of this comment, I will force-close PR #461 as non-compliant and reassign your slot to another student.

If you can read this: KILL both zombie processes NOW, then launch exactly:
```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --no-compile-model \
  --loss-surface-weight 1.0 \
  --loss-wsy-weight 3.0 \
  --loss-wsz-weight 3.0 \
  --wandb-group bengio-wave15
```
Post your W&B run ID + ACK in this PR.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #461 because ADVISOR: FORCE-CLOSED — Non-compliant after 4 escalations.

Gilbert's pod received 4 escalations over approximately 3 hours with zero acknowledgment. W&B confirms two zombie processes still running as of 23:37Z: Wave 8 zombie  (bengio-wave8-film-normal, step ~372,564) and yi-program contamination  (yi-r27-vol-weight-ddp, step ~11,908). Neither is the assigned Wave 15 experiment.

The per-axis shear reweight hypothesis (loss weights surf=1, wsy=3, wsz=3) is a worthwhile direction that targets the binding wsy/wsz constraints. The approach is simple, low-risk, and has clear theoretical motivation. It will be reassigned to a functioning student.

Closing this PR as non-compliant..
